lightningd/options.c

```diff
@@ -672,7 +672,19 @@ static void check_config(struct lightningd *ld)
 
 	if (ld->use_proxy_always && !ld->proxyaddr)
 		fatal("--always-use-proxy needs --proxy");
-}
+
+	if (ld->proxyaddr && (!ld->use_proxy_always || ld->config.use_dns)) {
+		if (tal_count(ld->proposed_wireaddr) != 0
+	    && some_tor_addresses(ld->proposed_wireaddr)) {
```

Comment:
Indentation :-)

Reply:
> What do you think of disabling DNS if --use-always-proxy=true ?


aktualy doing the follow PR right now 

`if (!daemon->use_proxy_always) 
      if (broken_resolver(daemon)) {
or so ...
		`